### PR TITLE
fix(Mesh): add mesh element data structure

### PR DESCRIPTION
**Pull request by panquez**

### CODE REVIEW

File: include/geode/mesh/core/mesh_element.h [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2019 - 2022 Geode-solutions. All rights reserved.
```
Comment (BotellaA) [inline include/geode/mesh/core/mesh_element.h:2]: Wrong header - copyright

File: include/geode/mesh/core/mesh_element.h [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+#include <geode/basic/uuid.h>
+
+#include <geode/mesh/common.h>
+
+namespace geode
+{
+    struct MeshElement
+    {
+        MeshElement( geode::uuid mesh_id_in, geode::index_t element_id_in )
```
Comment (BotellaA) [inline include/geode/mesh/core/mesh_element.h:15]: Useless geode::

File: include/geode/mesh/core/mesh_element.h [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        void serialize( Archive& archive )
+        {
+            archive.ext( *this, DefaultGrowable< Archive, MeshElement >{},
+                []( Archive& a, MeshElement& mesh_element ) {
+                    a.object( mesh_element.mesh_id );
+                    a.value4b( mesh_element.element_id );
+                } );
+        }
+
+        geode::uuid mesh_id;
```
Comment (BotellaA) [inline include/geode/mesh/core/mesh_element.h:40]: Geode:: to remove

File: include/geode/mesh/core/mesh_element.h [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        geode::index_t element_id;
+
+    private:
+        friend class bitsery::Access;
+        MeshElement() = default;
+    };
+
+    using ComponentVertex = MeshElement;
```
Comment (BotellaA) [inline include/geode/mesh/core/mesh_element.h:48]: Do we want strong type or alias is enough?

**Review by BotellaA — APPROVED** [review]

File: include/geode/mesh/core/mesh_element.h (hunk 4 above)
Comment (panquez) [reply include/geode/mesh/core/mesh_element.h]: I do not know. I suppose strong typing is better to avoid misusage.

Comment (BotellaA) [reply include/geode/mesh/core/mesh_element.h]: Maybe juste inherit from MeshElment + an alias to use the mother constructor. As you want.

**Pull request merged**